src/twisted/web/test/test_error.py

```diff
@@ -42,19 +42,29 @@ class ErrorTests(unittest.TestCase):
     def test_noMessageValidStatus(self):
         """
         If no C{message} argument is passed to the L{Error} constructor and the
-        C{code} argument is a valid HTTP status code, C{code} is mapped to a
-        descriptive string to which C{message} is assigned.
+        C{code} argument is a valid HTTP status code, C{message} is set to the
+        HTTP reason phrase for C{code}.
         """
         e = error.Error(b"200")
         self.assertEqual(e.message, b"OK")
+        self.assertEqual(str(e), "200 OK")
 
-    def test_noMessageInvalidStatus(self):
+    def test_noMessageForStatus(self):
         """
         If no C{message} argument is passed to the L{Error} constructor and
-        C{code} isn't a valid HTTP status code, C{message} stays L{None}.
+        C{code} isn't a known HTTP status code, C{message} stays L{None}.
         """
-        e = error.Error(b"InvalidCode")
+        e = error.Error(b"999")
         self.assertEqual(e.message, None)
+        self.assertEqual(str(e), "999")
+
+    def test_invalidStatus(self):
+        """
+        If C{code} isn't plausibly an HTTP status code (i.e., composed of
+        digits) it is rejected with L{ValueError}.
+        """
+        with self.assertRaises(ValueError):
+            error.Error(b"InvalidStatus")
```

Comment:
Maybe something like this, to also check that we don't get a any ValueError

```suggestion
    error = self.assertRaises(ValueError, error.Error, b"InvalidStatus")
    self.assertEqual("Not a valid HTTP status code: InvalidStatus", error.args[0])
```

Reply:
I am -1 for using the context manager for `assertRaises` as it looks there are people who will add assertions inside the context manager, and will think that all is good, when in fact the assertion code is never executed.

I hope that the code coverage can help identity these types of errors, so I don't think we should enforce this.

is just a nice to have.

Reply:
I appreciate this but as the suggested change is not correct and I'm not at my workstation I'm going to lea e it as-is.